rust/src/wrapper/adv.rs

```diff
@@ -0,0 +1,440 @@
+//! Advertisements
+
+use crate::wrapper::{
+    assigned_numbers::{COMPANY_IDS, SERVICE_IDS},
+    core::{Uuid128, Uuid16, Uuid32},
+};
+use itertools::Itertools;
+use nom::{combinator, multi, number};
+use std::fmt;
+use strum::IntoEnumIterator;
+
+/// The numeric code for a common data type.
+///
+/// For known types, see [CommonDataType], or use this type directly for non-assigned codes.
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Hash)]
+pub struct CommonDataTypeCode(u8);
+
+impl From<CommonDataType> for CommonDataTypeCode {
```

Comment:
I could go either way. The concept in the spec is called "common data type" so if we don't use that name, then the intent might be less clear. OTOH if you're using a number that's not in the spec, then it's the same _sort_ of thing, but not _exactly_ that.

Reply:
Or maybe make the enum `AssignedCommonDataType`?

Reply:
I think the enum name `CommonDataType` makes sense given that's the exact terminology in "Assigned Numbers". Personally I lean toward `DataTypeCode` for the isolated u8, since it makes it clear that the code within may not necessarily be a "Common Data Type".

But I think leaving it as-is would be good too.

Reply:
I'm uneasy about calling it something that implies it's usable for _any_ kind of type code so I'll leave it for now, but I'm happy to break API aggressively in the quest for better names...